controller/appcontroller.go

```diff
@@ -888,11 +890,13 @@ func (ctrl *ApplicationController) refreshAppConditions(app *appv1.Application)
 			errorConditions = append(errorConditions, appv1.ApplicationCondition{
```

Comment:
The `refreshAppConditions` method is executed periodically. So if the application has an invalid spec the time will be constantly changing. Instead of explicitly setting time I would suggest different approach: 
- change `Time` field to pointer (`*metav1.Time`)
- update `SetConditions` method to set condition to `time.Now()` unless app has the condition of the same type and time already set.

Reply:
Good catch, will address, thanks.